### PR TITLE
Support handling query strings

### PR DESCRIPTION
**Pull request by malexdev**
Handling query strings is simple, but it seemed like a good thing to add to the project.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -6,14 +6,40 @@ const resumer = require('resumer')
...
+  const fn = async req => {
+    const q = await query(req)
+    return q.test
+  }
+
+  const qs = {test: 'test'}
+  const url = await getUrl(fn)
+  const res = await request({url, qs})
+
+  t.deepEqual(res, 'test')
```
Comment (floatdrop) [inline test/index.js]: You can just call `t.deepEqual` in req handler.

File: test/index.js [hunk 2]
```diff
@@ -6,14 +6,40 @@ const resumer = require('resumer')
...
+  const fn = async req => {
+    const q = await query(req)
+    return {a: q.a, b: q.b, c: q.c}
+  }
+
+  const qs = {a: 'a', b: 'b', c: 'c'}
+  const url = await getUrl(fn)
+  const res = await request({url, qs})
+
+  t.deepEqual(res, JSON.stringify({a: 'a', b: 'b', c: 'c'}))
```
Comment (floatdrop) [inline test/index.js]: Here too – `JSON.stringify` is not good approach.

File: lib/server.js [hunk 3]
```diff
@@ -68,6 +69,10 @@ exports.json = (req, {limit = '1mb'} = {}) => Promise.resolve().then(() => {
   })
 })
 
+exports.query = req => Promise.resolve().then(() => {
```
Comment (floatdrop) [inline lib/server.js]: There is no need for `Promise` here – no async work is done.

**Review by floatdrop — CHANGES_REQUESTED** [review]

File: lib/server.js (hunk 3 above)
Comment (malexdev) [reply lib/server.js]: Gotcha. Thought we might want to keep it the same as the `.json` API so made it use the promise. Makes sense to make it sync though.

File: test/index.js [hunk 4]
```diff
@@ -6,14 +6,38 @@ const resumer = require('resumer');
...
 
   return listen(srv);
 };
 
+test('support query string', async t => {
+  const qs = {test: 'test'}
+
+  const fn = async req => {
+    const q = query(req)
+    t.deepEqual(q, qs)
```
Comment (floatdrop) [inline test/index.js]: With latest fix (7.2.2) – you need to return here something, otherwise request will hang.

File: test/index.js [hunk 5]
```diff
@@ -6,14 +6,38 @@ const resumer = require('resumer');
...
+  const url = await getUrl(fn)
+  await request({url, qs})
+})
+
+test('support query string with multiple options', async t => {
+  const qs = {a: 'a', b: 'b', c: 'c'}
+
+  const fn = async req => {
+    const q = query(req)
+    t.deepEqual(q, qs)
```
Comment (floatdrop) [inline test/index.js]: And here too.

**Review by floatdrop — CHANGES_REQUESTED** [review]

**Review by floatdrop — COMMENTED** [review]
`url.parse` is not quite get things right – maybe it is better to use [`query-string`](https://github.com/sindresorhus/query-string) module or look at [URL class](https://nodejs.org/api/url.html#url_the_whatwg_url_api). Anyways parsing query string is trivial task, should it have own helper?

**Comment by malexdev** [comment]
If you don't think it would be a good fit for micro that's fine, I just thought it'd be helpful to have built in.  Also, what is it that `url.parse` doesn't get right? I haven't seen any issues with it so far.

Not trying to be combative, I'm open to whatever you guys want to do. If you don't want querystring parsing built into Micro that's fine, just reject the PR. If you prefer to use the `query-string` module, that's fine, I'll modify my PR.

**Comment by floatdrop** [comment]
@malexdev standart module does not support arrays in query or nested objects for example:

```js
url.parse('http://google.com/?query=1&a[0]=1&a[1]=1', true).query
// { query: '1', 'a[0]': '1', 'a[1]': '1' }
```

I can not merge or reject PR, just judging by #174/#184 this kind of PR are preferred as readme sections or separate modules. It kinda tough way, but keeps code and package small and shiny.

**Comment by leo** [comment]
Thank you for your effort, guys! But this is not something we'd like to see built-in. The point of `micro` is to only provide very little (the most important) abstractions for microservices.

**Pull request closed without merging**